src/Microsoft.NetCore.Analyzers/Core/MicrosoftNetCoreAnalyzersResources.resx

```diff
@@ -1212,4 +1212,13 @@
   <data name="DoNotUseCreateEncryptorWithNonDefaultIVDescription" xml:space="preserve">
     <value>Symmetric encryption should always use a non-repeatable initialization vector to prevent dictionary attacks.</value>
   </data>
+  <data name="DoNotCreateTaskCompletionSourceWithWrongArgumentsDescription" xml:space="preserve">
+    <value>Argument '{0}' needs to be of type TaskCreationOptions enum, not of TaskContinuationsOptions enum.</value>
```

Comment:
Rename this resource string to `DoNotCreateTaskCompletionSourceWithWrongArgumentsMessage` and change contents to:

```suggestion
    <value>Argument '{0}' passed to TaskCompletionSource constructor must be of type TaskCreationOptions enum, not TaskContinuationsOptions enum.</value>
```